src/main/java/mate/academy/internetshop/dao/jdbc/BucketDaoJdbcImpl.java

```diff
@@ -0,0 +1,167 @@
+        super(connection);
+    }
+
+
+    @Override
+    public Bucket create(Bucket bucket) {
+        String query = "INSERT INTO buckets(user_id) VALUES(?);";
+        try (PreparedStatement stmt = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            stmt.setLong(1, bucket.getUserId());
+            stmt.executeUpdate();
+            ResultSet resultSet = stmt.getGeneratedKeys();
+            resultSet.next();
+            Long bucketId = resultSet.getLong(1);
+            bucket.setId(bucketId);
+            setBucketItems(bucket);
+        } catch (SQLException e) {
+            logger.error("couldn't creat bucket " + bucket, e);
+        }
+        return bucket;
+    }
+
+    private void setBucketItems(Bucket bucket) {
+        String query = "INSERT INTO buckets_items(bucket_id,item_id) VALUES(?,?);";
+        try (PreparedStatement stmt = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            for (Item item : bucket.getItems()) {
+                stmt.setLong(1, bucket.getId());
+                stmt.setLong(2, item.getId());
+                stmt.executeUpdate();
+            }
+        } catch (SQLException e) {
+            logger.error("couldn't set bucket's items " + bucket, e);
+        }
+    }
+
+    @Override
+    public Optional<Bucket> get(Long bucketId) {
+        String query = "SELECT * FROM buckets " +
+                "WHERE bucket_id =? ";
+        try (PreparedStatement stmt = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            stmt.setLong(1, bucketId);
+            ResultSet rs = stmt.executeQuery();
+            while (rs.next()) {
+                Long user_id = rs.getLong("user_id");
+                Bucket bucket = new Bucket();
+                bucket.setItems(getBucketItems(bucketId));
+                bucket.setId(bucketId);
+                bucket.setUserId(user_id);
+                return Optional.of(bucket);
+            }
+        } catch (SQLException e) {
+            logger.warn("Can't get bucket with id = " + bucketId, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Bucket> getByUserId(Long userId) {
+        String query = "SELECT * FROM buckets " +
+                "WHERE user_id =? ";
+        try (PreparedStatement stmt = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            stmt.setLong(1, userId);
+            ResultSet rs = stmt.executeQuery();
+            while (rs.next()) {
+                Long bucket_id = rs.getLong("bucket_id");
+                Bucket bucket = new Bucket();
+                bucket.setItems(getBucketItems(bucket_id));
+                bucket.setId(bucket_id);
+                bucket.setUserId(userId);
+                return Optional.of(bucket);
+            }
+        } catch (SQLException e) {
+            logger.warn("Can't find bucket with user_id = " + userId, e);
```

Comment:
```suggestion
            logger.error("Can't find bucket with user_id = " + userId, e);
```
переглягути всі catch блоки і замінити в них warn на error